flocker/node/_docker.py

```diff
@@ -527,6 +528,50 @@ def _address_in_use(self, apierror):
             return None
         return AddressInUse(address=(ip, int(port)), apierror=apierror)
 
+    def _image_data(self, image, container, state):
+        """
+        Supply data about an image, by either inspecting it or returning
+        cached data if available.
+
+        :param unicode image: The ID of the image.
+
+        :param unicode container: The ID of the container for which this
+            image is being used.
+
+        :param unicode state: The state of the container.
+
+        :return: ``dict`` representing data about the image properties.
+        """
+        # This caching technique isn't thread-safe, but allowable
+        # in this case because a) the data is for our purposes
+        # immutable, so two writes of same key isn't a problem and
+        # b) we never remove entries.
```

Comment:
"never remove entries" strikes me as a memory usage problem waiting to happen.  These values probably do actually need to be invalidated, eventually, somehow.

Lacking any specific information about what the realistic usage patterns for this image data is, the simpler the solution the better here.  Something like an LRU is probably a good start (and there's no shortage of implementations of LRU caches for Python - eg http://docs.repoze.org/lru/ seems to be somewhat popular, and it has a decorator form that may even simplify the rest of this `_image_data` method).